pages/mercury_tour_page.py

```diff
@@ -1,73 +1,142 @@
-
-    def flights_ver_click(self):
-        # Move to flights vertical
-        self.driver.find_element_by_link_text('Flights').click()
-
-    def pass_count(self, passcount):
-        passenger = self.driver.find_element_by_name('passCount')
-        Select(passenger).select_by_visible_text(passcount)
-
-    def from_port(self, fromport):
-        departing_port = self.driver.find_element_by_name('fromPort')
-        Select(departing_port).select_by_visible_text(fromport)
-
-    def from_month(self, frommonth):
-        depart_month = self.driver.find_element_by_name('fromMonth')
-        Select(depart_month).select_by_visible_text(frommonth)
-
-    def from_day(self, fromday):
-        depart_day = self.driver.find_element_by_name('fromDay')
-        Select(depart_day).select_by_visible_text(fromday)
-
-    def to_port(self, toport):
-        arriving_in = self.driver.find_element_by_name('toPort')
-        Select(arriving_in).select_by_visible_text(toport)
-
-    def to_month(self, tomonth):
-        return_month = self.driver.find_element_by_name('toMonth')
-        Select(return_month).select_by_visible_text(tomonth)
-
-    def to_day(self, today):
-        return_date = self.driver.find_element_by_name('toDay')
-        Select(return_date).select_by_visible_text(today)
-
-    def sevice_cred(self):
-        self.driver.find_element_by_css_selector('input[value="Business"]').click()
-
-    def airline_name_select(self, airline):
-        airline_name = self.driver.find_element_by_css_selector('select[name="airline"]')
-        Select(airline_name).select_by_visible_text(airline)
-
-    def flight_det_sub(self):
+        """
+        Submit username and password in fields and submit
+        Returned next page
+        """
+        WebDriverWait(self.driver, 10).until(
+            EC.presence_of_element_located((By.NAME, 'userName'))
+        ).send_keys(username)
+        WebDriverWait(self.driver, 10).until(
+            EC.presence_of_element_located((By.NAME, 'password'))
+        ).send_keys(password)
+        self.driver.find_element_by_name('submit').click()
+        WebDriverWait(self.driver, 10).until(
+            EC.presence_of_element_located((By.CSS_SELECTOR, 'img[src="images/banner2.gif"]'))
+        )
+
+    def open_flight_vertical_by_clicking(self):
+        """
+        Move to flights vertical
+        """
+        WebDriverWait(self.driver, 10).until(
+            EC.presence_of_element_located((By.LINK_TEXT, 'Flights'))
+        ).click()
+        footer = WebDriverWait(self.driver, 10).until(
+            EC.visibility_of_element_located((By.CSS_SELECTOR, 'div.footer'))
+        ).text
+        return footer
+
+    def select_passenger(self, passcount):
+        """
+        Enter any passcount in passenger field
+        """
+        Select(WebDriverWait(self.driver, 10).until(
+            EC.visibility_of_element_located((By.NAME, 'passCount'))
+        )).select_by_visible_text(passcount)
+        return passcount
```

Comment:
we shouldn't return the values we're entering in input fields (as we don't need to assert them in test).

Reply:
Done